Use single GPU mining thread when --opencl-device is chosen

Genoil opened the pull request:
bugfix

gavofyork commented on libethash-cl/ethash_cl_miner.cpp:
```diff
@@ -29,6 +29,7 @@
 #include <vector>
 #include <libethash/util.h>
 #include <libethash/ethash.h>
+#include <libdevcore/Log.h>
```
best not to make this require libdevcore

Genoil commented:
Ok I removed it and replaced it with cout. The reason I did it was mostly this:
```
Benchmarking on platform: { "platform": "NVIDIA CUDA", "device": "GeForce GTX 78
0", "version": "OpenCL 1.2 CUDA" }
Preparing DAG...
  D  12:33:41|main  pause took 0
  i  12:33:41|gpuminer0  Entering work loop...
 U sii n g1 p2l:a3t3f:o4r1m|:m aNiVnI D ISAp aCwUnDiAn
gU sgipnugm idneevri0ce: GeForce GTX 780(OpenCL 1.2 CUDA)

  D  12:33:41|main  kickOff took 0.007
  D  12:33:41|main  pause took 0
  i  12:33:41|main  Spawning gpuminer1
  i  12:33:41|gpuminer1  Entering work loop...
Using platform: N V IDD I A1 2C:U3D3A:
41|main  kickOff took 0.002
Warming up...
Using device: GeForce GTX 750 Ti(OpenCL 1.2 CUDA)
Trial 1... 20534613
Trial 2... 20601393
Trial 3... 20353066
Trial 4... 20615122
Trial 5... 20527770
  i  12:34:09|main  Stopping gpuminer0
  i  12:34:09|gpuminer0  Finishing up worker thread...
  i  12:34:09|main  Terminating gpuminer0
  i  12:34:12|gpuminer1  Finishing up worker thread...
  i  12:34:12|main  Stopping gpuminer1
  i  12:34:12|main  Terminating gpuminer1
min/mean/max: 20353066/20526392/20615122 H/s
inner mean: 20523193 H/s
Phoning home to find world ranking...
Ranked: 5 of all benchmarks.
```
It looks like two threads are outputting to stdout at the same time, resulting in complete gibberisch. Using the core Logger solved that. Is this a Windows thing or does Linux also have this?

gavofyork commented:
yeah - that's probably an issue on all platforms.

the real way to fix is to avoid any console output from the library and have decent diagnostics upstream.
